### PR TITLE
feat: show share api

### LINKED ISSUE #6754: show share.
add `show_share` in ShareApi, return share meta and share account meta.

arg:

```
#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
pub struct ShowShareReq {
    pub share_name: ShareNameIdent,
}

#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
pub struct ShowShareReply {
    pub share_name: ShareNameIdent,

    pub share_id: u64,

    pub share_meta: ShareMeta,

    pub share_account_meta: Vec<ShareAccountMeta>,
}
```

### PR DESCRIPTION
**Pull request by ClSlaid**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR implements the` Show` operation in meta's share API.

Fixes #6754

### CODE REVIEW

**Review by lichuang — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
Run cargo fmt --all -- --check
Diff in /workspace/common/meta/api/src/share_api_test_suite.rs at line 21:
 use common_meta_app::schema::TableMeta;
 use common_meta_app::schema::TableNameIdent;
 use common_meta_app::share::AddShareAccountReq;
-use common_meta_app::share::ShowShareReq;
 use common_meta_app::share::CreateShareReq;
 use common_meta_app::share::DropShareReq;
 use common_meta_app::share::GrantShareObjectReq;
Diff in /workspace/common/meta/api/src/share_api_test_suite.rs at line 32:
 use common_meta_app::share::ShareGrantObjectName;
 use common_meta_app::share::ShareGrantObjectPrivilege;
 use common_meta_app::share::ShareNameIdent;
+use common_meta_app::share::ShowShareReq;
 use common_tracing::tracing;
 use enumflags2::BitFlags;
Diff in /workspace/common/meta/app/src/share/mod.rs at line 15:
 #[allow(clippy::module_inception)]
 mod share;
-pub use share::ShowShareReq;
-pub use share::ShowShareReply;
 pub use share::AddShareAccountReply;
 pub use share::AddShareAccountReq;
 pub use share::CreateShareReply;
Diff in /workspace/common/meta/app/src/share/mod.rs at line 42:
 pub use share::ShareInfo;
 pub use share::ShareMeta;
 pub use share::ShareNameIdent;
+pub use share::ShowShareReply;
+pub use share::ShowShareReq;
```

**Review by drmingdrmer — APPROVED** [review]
Looks good to me:D

File: tests/logictest/http_connector.py [hunk 1]
```diff
@@ -52,8 +52,7 @@ def format_result(results):
             if buf == "":
                 buf = str(item)
             else:
-                buf = buf + " " + str(
-                    item)  # every item seperate by space
+                buf = buf + " " + str(item)  # every item seperate by space
```
Comment (lichuang) [inline tests/logictest/http_connector.py:55]: this unrelated pr can be rollback?

Comment (ClSlaid) [reply tests/logictest/http_connector.py:55]: this is changed by `make lint`, I'll roll it back.

**Pull request merged**